shared/src/types/vote_extensions/validator_set_update.rs

```diff
@@ -0,0 +1,297 @@
+            let data = Vext {
+                validator_addr,
+                voting_powers,
+                epoch,
+            };
+            extensions.push(SignedVext::new_from(data, signature));
+        }
+        extensions
+    }
+
+    /// Returns an Ethereum ABI encoded string with the
+    /// params to feed to the Ethereum bridge smart contracts.
+    pub fn abi_params(&self) -> String {
+        todo!()
+    }
+}
+
+/// Represents a [`Vext`] signed by some validator, with
+/// an Ethereum key.
+pub type SignedVext = Signed<Vext, SerializeWithAbiEncode>;
+
+impl SignedVext {
+    /// Serialize a [`Vext`] to be signed.
+    fn serialize_vext(ext: &Vext) -> [u8; 32] {
+        AbiEncode::signed_keccak256(&[
+            Token::String("updateValidatorsSet".into()),
+            Token::FixedBytes(
+                ext.voting_powers.get_bridge_hash(ext.epoch).to_vec(),
+            ),
+            Token::FixedBytes(
+                ext.voting_powers.get_governance_hash(ext.epoch).to_vec(),
+            ),
+            epoch_to_token(ext.epoch),
+        ])
+    }
+
+    /// Sign this [`Vext`] with an Ethereum key.
+    ///
+    /// For more information, check the Ethereum bridge smart contract code:
+    ///   - <https://github.com/anoma/ethereum-bridge/blob/main/contracts/contract/Bridge.sol#L186>
+    pub fn new_abi_encoded(keypair: &common::SecretKey, ext: Vext) -> Self {
+        let to_sign = Self::serialize_vext(&ext);
+        let sig = common::SigScheme::sign(keypair, &to_sign);
+        Self::new_from(ext, sig)
+    }
+
+    /// Verify the signature of a [`Vext`], signed by some
+    /// Ethereum key.
+    pub fn verify_abi_encoded(
+        &self,
+        pk: &common::PublicKey,
+    ) -> Result<(), VerifySigError> {
+        let bytes = Self::serialize_vext(&self.data);
+        common::SigScheme::verify_signature_raw(pk, &bytes, &self.sig)
+    }
+}
+
+/// Represents a validator set update, for some new [`Epoch`].
+#[derive(
+    Eq, PartialEq, Clone, Debug, BorshSerialize, BorshDeserialize, BorshSchema,
+)]
+pub struct Vext {
+    /// The addresses of the validators in the new [`Epoch`],
+    /// and their respective voting power.
+    ///
+    /// When signing a [`Vext`], this [`VotingPowersMap`] is converted
+    /// into two arrays: one for its keys, and another for its
+    /// values. The arrays are sorted in descending order based
+    /// on the voting power of each validator.
+    pub voting_powers: VotingPowersMap,
+    /// TODO: the validator's address is temporarily being included
+    /// until we're able to map a Tendermint address to a validator
+    /// address (see https://github.com/anoma/namada/issues/200)
+    pub validator_addr: Address,
+    /// The new [`Epoch`].
+    ///
+    /// Since this is a monotonically growing sequence number,
+    /// it is signed together with the rest of the data to
+    /// prevent replay attacks on validator set updates.
+    pub epoch: Epoch,
```

Comment:
It also allows us to look up with valset is needed to verify signatures of user txs coming to the eth side of the bridge.